components/builder-graph/src/target_graph.rs

```diff
@@ -55,7 +55,11 @@ impl TargetGraph {
         }
     }
 
-    pub fn graph_mut(&mut self, target_str: &str) -> Option<&mut PackageGraph> {
+    #[allow(clippy::borrowed_box)]
+    // Clippy wants
+    // pub fn graph_mut(&mut self, target_str: &str) -> Option<&mut dyn PackageGraphTrait>
+    // But then lifetime issues intrude if we follow the as_ref pattern above.
+    pub fn graph_mut(&mut self, target_str: &str) -> Option<&mut Box<dyn PackageGraphTrait>> {
```

Comment:
Using the signiture `pub fn graph_mut(&mut self, target_str: &str) -> Option<&mut (dyn PackageGraphTrait + 'static)>` fixes the clippy and lifetime errors. The `&` case has the `'static` lifetime as well, but for some reason the compiler is better at infering it in that case :shrug:.

Reply:
`Box` with a trait object implicitly has the `'static` lifetime. This [thread](https://users.rust-lang.org/t/box-with-a-trait-object-requires-static-lifetime/35261) has some more details if you are interested.